iocage/cli/get.py

```diff
@@ -23,97 +23,82 @@
-            })
-
-    if not recursive:
-        if prop == "state":
-            state = ioc.IOCage(jail).get(prop)
-
-            ioc_common.logit({
-                "level"  : "INFO",
-                "message": state
-            })
-        elif plugin:
-            _plugin = ioc.IOCage(jail, skip_jails=True).get(prop, plugin=True)
-
-            ioc_common.logit({
-                "level"  : "INFO",
-                "message": _plugin
-            })
-        elif prop == "all":
-            props = ioc.IOCage(jail, skip_jails=True).get(prop)
-
-            for p, v in props.items():
-                ioc_common.logit({
-                    "level"  : "INFO",
-                    "message": f"{p}:{v}"
-                })
-        else:
-            p = ioc.IOCage(jail, skip_jails=True).get(prop)
-
-            ioc_common.logit({
-                "level"  : "INFO",
-                "message": p
-            })
+
+    logger.print_level = log_level
+
+    if _pool is True:
+        try:
+            print(host.datasets.active_pool.name)
+        except:
+            print("No active pool found")
+        exit(1)
+
+    if jail == "":
+        jail_identifier = prop
+        prop = ""
     else:
-        jail_list = ioc.IOCage().get(prop, recursive=True)
-
-        # Prints the table
-        if header:
-            jail_list.insert(0, ["NAME", f"PROP - {prop}"])
-            # We get an infinite float otherwise.
-            table.set_cols_dtype(["t", "t"])
-            table.add_rows(jail_list)
-            ioc_common.logit({
-                "level"  : "INFO",
-                "message": table.draw()
-            })
-        else:
-            for jail in jail_list:
-                ioc_common.logit({
-                    "level"  : "INFO",
-                    "message": "\t".join(jail)
-                })
+        jail_identifier = jail
+
+    jail_identifier = jail
+    jail = Jail.Jail(
+        jail_identifier,
+        host=host,
+        logger=logger
+    )
+
+    if not jail.exists:
+        logger.error("Jail '{jail}' does not exist")
+        exit(1)
+
+    if _all is True:
+        prop = None
+
+    if prop == "all":
+        prop = None
+
+    if (prop is None) and (jail_identifier == "") and not _all:
+        logger.error("Missing arguments property and jail")
+        raise Exception("foo")
+        exit(1)
+    elif (prop is not None) and (jail_identifier == ""):
+        logger.error("Missing argument property name or -a/--all argument")
+        exit(1)
+
+    if prop:
+        try:
+            print_property(prop, jail.config.__getattr__(prop, string=True))
+            return
+        except:
+            raise
+            logger.error(f"Unknown property '{prop}'")
+
+    for key in jail.config.all_properties:
+        if (prop is None) or (key == prop):
+            value = jail.config.__getattr__(key, string=True)
+            print_property(key, value)
+
+
+def print_property(key, value):
+    key = str(key)
```

Comment:
The str method of the property should be invoked autmatically when `print(f"{key}:{value}")` is evaluated is my point.

Let me provide an example:
```
>>> class Dummy():
...     def __str__(self):
...         return "__class__:fi fo fi fum"
...
...
>>> d = Dummy()
>>> print(f"Here it is: {d}")
>>> Here it is: __class__:fi fo fi fum

>>>
```